SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp

```diff
@@ -116,3 +134,65 @@ void USpatialDispatcher::ProcessOps(Worker_OpList* OpList)
 		}
 	}
 }
+
+bool USpatialDispatcher::IsExternalSchemaOp(Worker_Op* Op) {
+	Worker_ComponentId ComponentId = GetComponentId(Op);
+	return  SpatialConstants::MIN_EXTERNAL_SCHEMA_ID <= ComponentId && ComponentId <= SpatialConstants::MAX_EXTERNAL_SCHEMA_ID;
+}
+
+void USpatialDispatcher::ProcessExternalSchemaOp(Worker_Op* Op) {
+	auto TryUserCallback = [&](Worker_ComponentId ComponentId, TFunction<void(UOpCallbackTemplate*)>& OpCallback) {
+		if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(ComponentId))
+		{
+			UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+			OpCallback(UserCallback);
+		}
+	};
+
+	Worker_ComponentId ComponentId = GetComponentId(Op);
+	TFunction<void(UOpCallbackTemplate*)> UserCallback;
+	switch (Op->op_type)
+	{
+	case WORKER_OP_TYPE_ADD_COMPONENT:
+		UserCallback = [&](UOpCallbackTemplate* UserCallback) {UserCallback->OnAddComponent(Op->add_component); };
+		break;
+	case WORKER_OP_TYPE_REMOVE_COMPONENT:
+		UserCallback = [&](UOpCallbackTemplate* UserCallback) {UserCallback->OnRemoveComponent(Op->remove_component); };
+		break;
+	case WORKER_OP_TYPE_COMPONENT_UPDATE:
+		UserCallback = [&](UOpCallbackTemplate* UserCallback) {UserCallback->OnComponentUpdate(Op->component_update); };
+		break;
+	case WORKER_OP_TYPE_AUTHORITY_CHANGE:
+		UserCallback = [&](UOpCallbackTemplate* UserCallback) {UserCallback->OnAuthorityChange(Op->authority_change); };
```

Comment:
Could we also keep the view up to date with this authority change - ie. `StaticComponentView->OnAuthorityChange(Op->authority_change);`

Reply:
I think there was a reason against storing this component data in the StaticComponentView but I can't remember, will look tmmo.